js-wixcode-sdk/$w/repeater/guides/The Lifecycle of Repeated Items.md

```diff
@@ -0,0 +1,132 @@
+---
+articleType: concept
+---
+# The Lifecycle of Repeated Items
+
+
+When you work with a [repeater](https://dev.wix.com/docs/velo/velo-only-apis/$w/repeater/introduction) in code, it's important to understand how the items in the repeater are created, updated, and deleted. 
+
+## Connect data
+
+A repeater's data is an array of objects, which you get or set using the repeater's `data` property. Each object in this array must have a unique `_id` property that links the object to a specific repeated item in the repeater. The `_id` value can include only alphanumeric characters and hyphens (`-`). Besides `_id`, each object in the data array can include any other properties you need.
+
+The repeater doesn't automatically connect data in the elements inside each repeated item. You control how to connect data in those elements by using the `onItemReady()`, `onItemRemoved()`, `forItems()`, and `forEachItem()` callback functions.
+
+
+### Example:
+
+```javascript
+const data = [
+  { _id: "1", name: "Alice" },
+  { _id: "2", name: "Bob" }
+];
+
+$w.onReady(function () {
+  $w('#myRepeater').data = data;
+  $w('#myRepeater').onItemReady(($item, itemData) => {
+    $item('#nameText').text = itemData.name;
+  });
+});
+```
+
+## Create new items
+
+When you first add a repeater to your page, it displays repeated items using the design and default values from its [item template](https://github.com/wix-incubator/wix-code-docs/pull/3381/Repeated%20Item%20Template.md). At this stage, no custom data is shown.
+
+To display your own data, set the repeater’s `data` property to an array of objects, each with a unique `_id`. When you do this, the repeater creates new items for each object in the array.
+
+As each new item is created, the `onItemReady()` event handler runs. This is where you write code to update the elements inside each repeated item with the specific data you want to display, replacing the template's default values.
```

Comment:
Let's not use "update" in our "create" section. Let's stick with "populate" as suggested above.

```suggestion
As each new item is created, the `onItemReady()` event handler runs. This is where you write code to populate the elements inside each repeated item with the specific data you want to display, replacing the template's default values.
```